src/sentry/testutils/thread_leaks/assertion.py

```diff
@@ -0,0 +1,53 @@
+"""Utility used to regression-proof our thread-leak fixes.
+
+It turns out due to various indirections, just showing the name and exact
+"target" of a thread isn't enough to find where/how it was spawned. This code
+does a bit of work to record and show exactly where the Thread came from, which
+proved essential when working to fix these things.
+"""
+
+import threading
+import traceback
+from contextlib import contextmanager
+from threading import Thread
+from typing import Any
+
+from . import sentry
+from .diff import diff
+
+
+class ThreadLeakAssertionError(AssertionError):
+    pass
+
+
+@contextmanager
+def threading_remembers_where():
+    """Smuggle a ._where StackSummary attribute onto each Thread construction."""
+    __init__ = Thread.__init__
+
+    def patched__init__(self: Thread, *a: Any, **k: Any) -> None:
+        setattr(self, "_where", traceback.extract_stack())
+        __init__(self, *a, **k)
+
+    from unittest import mock
```

Comment:
This would result in overhead when the context manager is used a lot in tests, can this be moved to top of file?

Reply:
fixed.